### PR TITLE
QR-Code-scanner: integrate QUIRC library, implement QrDecoder, drop ZBar

### PR DESCRIPTION
**Pull request by xiphon**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -131,12 +131,8 @@ message(STATUS "OpenSSL: Version ${OPENSSL_VERSION}")
 message(STATUS "OpenSSL: include dir at ${OPENSSL_INCLUDE_DIR}")
 message(STATUS "OpenSSL: libraries at ${OPENSSL_LIBRARIES} ${OPENSSL_SSL_LIBRARIES}")
 
-# Zbar (for QR scanner)
 if(WITH_SCANNER)
     add_definitions(-DWITH_SCANNER)
```
Comment (selsta) [inline CMakeLists.txt:135]: What’s the point of this?

**Comment by selsta** [comment]
I wasn’t able to scan a QR code on Mac, the camera loads fine but it does not recognise any QR codes.

This was in my warnings, not sure if related.
```
QObject::connect: No such signal QrScanThread::decoded(int, QString)
```

File: CMakeLists.txt (hunk 1 above)
Comment (xiphon) [reply CMakeLists.txt:135]: To optionally support building GUI without scanner support. Windows docker doesn't support it yet.

**Comment by xiphon** [comment]
> I wasn’t able to scan a QR code on Mac, the camera loads fine but it does not recognise any QR codes.
> 
> This was in my warnings, not sure if related.
> 
> ```
> QObject::connect: No such signal QrScanThread::decoded(int, QString)
> ```

Yep, thanks. Should be fixed now.

**Review by selsta — APPROVED** [review]

**Pull request merged**